web/src/components/filings-calendar/tax-access/TaxAccessStepTwo.tsx

```diff
@@ -29,12 +31,11 @@ import { Backdrop, CircularProgress } from "@mui/material";
 import { ReactElement, useState } from "react";
 
 interface Props {
-  isOpen: boolean;
-  close: () => void;
   onSuccess: () => void;
   moveToPrevStep: () => void;
   CMS_ONLY_fakeError?: "NONE" | "API" | "UNKNOWN";
   CMS_ONLY_fakeBusiness?: Business;
+  hadLegalStructureOnMount?: boolean;
```

Comment:
pebble - instead of this prop, we could use the legal structure value from useUserData

Reply:
This prop is here to track if the user had a legal structure when the component first mounts. There are two cases: Either the user starts from step 1, enters a business structure, hits "next" and should see a back button to go back to step 1 - or -  if they didn't have to go through step 1 and enter a legal structure in the first place, drop them into step 2 without showing a back button. Using the presence of the legalStructureId for the back button logic means it never shows up even in case 1.

Regardless... this whole 2-step process is going away in a future design update anyway, addressing this slightly clunky behavior. So I'm going to leave this logic as is while we're working with two steps!